fedora_messaging/cli.py

```diff
@@ -452,3 +458,34 @@ def record(exchange, queue_name, routing_key, app_name, limit, file):
     _consume(
         exchange, queue_name, routing_key, messages_recorder.collect_message, app_name
     )
+
+
+URL_TEMPLATE = "https://apps.fedoraproject.org/datagrepper/id?id={}&is_raw=true"
+
+
+@cli.command()
+@click.argument("message_id")
+@click.option("--datagrepper-url", help=_datagrepper_url, default=URL_TEMPLATE)
+def replay(message_id, datagrep_url):
+    """Replay a message from Datagrepper by its message ID"""
+    message_data = _get_message(message_id, datagrep_url)
+    if message_data:
+        try:
+            api.publish(Message.load_message(message_data))
+            click.echo(f"Message with ID {message_id} has been successfully replayed.")
+        except Exception as e:
+            click.ClickException(f"Failed to replay message: {e}")
+    else:
+        click.ClickException(f"Could not find message with ID {message_id}")
+
+
+def _get_message(message_id, datagrep_url):
+    """Fetch a message by ID from Datagreeper"""
+    url = datagrep_url.format(message_id)
+    try:
+        response = requests.get(url, timeout=5)
+        response.raise_for_status()
+        return response.json()
+    except requests.HTTPError as e:
+        click.ClickException(f"Failed to retrieve message from Datagrepper: {e}")
```

Comment:
You have to raise this exception, here you're just instantiating it and raising the original error.